### PR TITLE
enhancement(internal_metrics)!: Default to setting the host tag

### PR DESCRIPTION
**Pull request by jszwedko**
Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

File: src/sources/internal_metrics.rs [hunk 1]
```diff
@@ -73,11 +73,15 @@ impl SourceConfig for InternalMetricsConfig {
...
-        let host_key = self
+        let host_key = match self
             .tags
             .host_key
             .as_deref()
-            .and_then(|tag| (!tag.is_empty()).then(|| tag.to_owned()));
+            .unwrap_or_else(|| log_schema().host_key())
+        {
+            "" => None,
+            s => Some(s.to_owned()),
```
Comment (spencergilbert) [inline src/sources/internal_metrics.rs:83]: Don't love the single char name here but 🤷

**Review by spencergilbert — APPROVED** [review]
I remember this being an intentional choice - was it just avoiding breaking the existing behavior, or was there something more? I don't want us to unintentionally cause another issue.

File: website/content/en/highlights/2022-10-04-0-25-0-upgrade-guide.md [hunk 2]
```diff
@@ -40,6 +41,15 @@ After:
 mod(5, 2) == 1
 ```
 
+#### `internal_metrics` defaults to setting `host` tag {#internal_metrics-host_tag)
+
+The `internal_metrics` source now defaults to setting the host tag on emitted metrics. This behavior
+avoids issues when deploying multiple Vector instances publishing their metrics to the same sink
+without opting into this tag via `tags.host_key`. The default key for this the configured
```
Comment (neuronull) [inline website/content/en/highlights/2022-10-04-0-25-0-upgrade-guide.md:48]:
```suggestion
without opting into this tag via `tags.host_key`. The default key for this is the configured
```

**Review by neuronull — APPROVED** [review]
LGTM

**Pull request merged**